Using both `try_setter` and `setter(strip_option)` does not compile

Using both `try_setter` and `setter(trip_option)` results in the error expected enum `Option`, found struct `String`.

version: `0.12.0`
repro:
```rust
#[derive(derive_builder::Builder)]
struct Example {
    #[builder(try_setter, setter(strip_option))]
    value: Option<String>,
}
```
```
error[E0308]: mismatched types
   --> src/lib.rs:3:10
    |
3   | #[derive(Builder)]
    |          ^^^^^^^
    |          |
    |          expected enum `Option`, found struct `String`
    |          arguments to this enum variant are incorrect
    |
    = note: expected enum `Option<String>`
             found struct `String`
```
https://github.com/chanced/derive_builder_issue

--- a/derive_builder/CHANGELOG.md
+++ b/derive_builder/CHANGELOG.md
@@ -2,6 +2,9 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased] - 2023-04-06
+- Make try-setters inherit `strip_option` from `setter` for `try_setter`. Using these settings together previously caused a compile error  #284
+
 ## [0.12.0] - 2022-11-28
 - Produce error when `default` is used with `field(type = "...")` rather than silently ignoring `default` #269
 - Add support for `crate = "..."` to support re-export scenarios #274

--- a/derive_builder/tests/setter_custom.rs
+++ b/derive_builder/tests/setter_custom.rs
@@ -1,3 +1,5 @@
+use std::convert::TryInto;
+
 #[macro_use]
 extern crate pretty_assertions;
 #[macro_use]
@@ -16,6 +18,8 @@ struct SetterCustom {
     setter_custom_with_explicit_default: u32,
     #[builder(setter(custom = "true", strip_option))]
     setter_custom_with_strip_option: Option<u32>,
+    #[builder(try_setter, setter(custom = "true", strip_option))]
+    setter_custom_with_strip_option_try_setter: Option<u32>,
 }
 
 // compile test
@@ -44,6 +48,19 @@ impl SetterCustomBuilder {
         self.setter_custom_with_strip_option = Some(Some(6));
         self
     }
+    fn setter_custom_with_strip_option_try_setter(&mut self) -> &mut Self {
+        self.setter_custom_with_strip_option_try_setter = Some(Some(32));
+        self
+    }
+}
+
+struct TryIntoU32 {}
+impl TryInto<u32> for TryIntoU32 {
+    type Error = ();
+
+    fn try_into(self) -> Result<u32, Self::Error> {
+        Ok(32)
+    }
 }
 
 #[test]
@@ -58,6 +75,7 @@ fn setter_custom_defaults() {
             setter_custom_by_explicit_opt_out: 0,
             setter_custom_with_explicit_default: 4,
             setter_custom_with_strip_option: None,
+            setter_custom_with_strip_option_try_setter: None
         }
     );
 }
@@ -70,6 +88,7 @@ fn setter_custom_setters_called() {
         .setter_custom_by_explicit_opt_out(42)
         .setter_custom_with_explicit_default() // set to 43
         .setter_custom_with_strip_option() // set to 6
+        .setter_custom_with_strip_option_try_setter() // set to 32
         .build()
         .unwrap();
 
@@ -80,7 +99,8 @@ fn setter_custom_setters_called() {
             setter_custom_shorthand: 2,
             setter_custom_by_explicit_opt_out: 42,
             setter_custom_with_explicit_default: 43,
-            setter_custom_with_strip_option: Some(6)
+            setter_custom_with_strip_option: Some(6),
+            setter_custom_with_strip_option_try_setter: Some(32)
         }
     );
 }

--- a/derive_builder/tests/try_setter.rs
+++ b/derive_builder/tests/try_setter.rs
@@ -85,3 +85,9 @@ fn renamed() {
         .build()
         .expect("All fields were provided");
 }
+
+#[derive(Debug, PartialEq, Builder)]
+#[builder(try_setter, setter(into, strip_option))]
+struct MaybeIpsum {
+    pub source: Option<MyAddr>,
+}

--- a/derive_builder_core/src/setter.rs
+++ b/derive_builder_core/src/setter.rs
@@ -469,7 +469,7 @@ mod tests {
         );
     }
     #[test]
-    fn strip_option_try_into() {
+    fn strip_option_try_setter() {
         let ty = parse_quote!(Option<Foo>);
         let mut setter = default_setter!();
         setter.strip_option = true;

--- a/derive_builder_core/src/setter.rs
+++ b/derive_builder_core/src/setter.rs
@@ -160,7 +160,12 @@ impl<'a> ToTokens for Setter<'a> {
                 let try_ident = syn::Ident::new(&format!("try_{}", ident), Span::call_site());
 
                 let mut converted = quote! {converted};
+
                 if builder_field_is_option {
+                    converted = wrap_expression_in_some(crate_root, converted);
+                }
+
+                if stripped_option {
                     converted = wrap_expression_in_some(crate_root, converted);
                 }
 
@@ -458,6 +463,43 @@ mod tests {
                         ::db::export::core::option::Option::Some(value.into())
                     );
                     new
+                }
+            )
+            .to_string()
+        );
+    }
+    #[test]
+    fn strip_option_try_into() {
+        let ty = parse_quote!(Option<Foo>);
+        let mut setter = default_setter!();
+        setter.strip_option = true;
+        setter.try_setter = true;
+        setter.generic_into = true;
+        setter.field_type = BuilderFieldType::Optional(&ty);
+        assert_eq!(
+            quote!(#setter).to_string(),
+            quote!(
+                #[allow(unused_mut)]
+                pub fn foo<VALUE: ::db::export::core::convert::Into<Foo>>(
+                    &mut self,
+                    value: VALUE,
+                ) -> &mut Self {
+                    let mut new = self;
+                    new.foo = ::db::export::core::option::Option::Some(
+                        ::db::export::core::option::Option::Some(value.into()),
+                    );
+                    new
+                }
+                pub fn try_foo<VALUE: ::db::export::core::convert::TryInto<Foo>>(
+                    &mut self,
+                    value: VALUE,
+                ) -> ::db::export::core::result::Result<&mut Self, VALUE::Error> {
+                    let converted: Foo = value.try_into()?;
+                    let mut new = self;
+                    new.foo = ::db::export::core::option::Option::Some(
+                        ::db::export::core::option::Option::Some(converted),
+                    );
+                    Ok(new)
                 }
             )
             .to_string()

--- a/derive_builder_core/src/setter.rs
+++ b/derive_builder_core/src/setter.rs
@@ -160,11 +160,9 @@ impl<'a> ToTokens for Setter<'a> {
                 let try_ident = syn::Ident::new(&format!("try_{}", ident), Span::call_site());
 
                 let mut converted = quote! {converted};
-
                 if builder_field_is_option {
                     converted = wrap_expression_in_some(crate_root, converted);
                 }
-
                 if stripped_option {
                     converted = wrap_expression_in_some(crate_root, converted);
                 }

--- a/derive_builder_core/src/setter.rs
+++ b/derive_builder_core/src/setter.rs
@@ -476,28 +476,29 @@ mod tests {
         setter.try_setter = true;
         setter.generic_into = true;
         setter.field_type = BuilderFieldType::Optional(&ty);
+        #[rustfmt::skip]
         assert_eq!(
             quote!(#setter).to_string(),
             quote!(
                 #[allow(unused_mut)]
                 pub fn foo<VALUE: ::db::export::core::convert::Into<Foo>>(
                     &mut self,
-                    value: VALUE,
+                    value: VALUE
                 ) -> &mut Self {
                     let mut new = self;
                     new.foo = ::db::export::core::option::Option::Some(
-                        ::db::export::core::option::Option::Some(value.into()),
+                        ::db::export::core::option::Option::Some(value.into())
                     );
                     new
                 }
                 pub fn try_foo<VALUE: ::db::export::core::convert::TryInto<Foo>>(
                     &mut self,
-                    value: VALUE,
+                    value: VALUE
                 ) -> ::db::export::core::result::Result<&mut Self, VALUE::Error> {
                     let converted: Foo = value.try_into()?;
                     let mut new = self;
                     new.foo = ::db::export::core::option::Option::Some(
-                        ::db::export::core::option::Option::Some(converted),
+                        ::db::export::core::option::Option::Some(converted)
                     );
                     Ok(new)
                 }

